src/ringmesh/geomodel/geomodel_builder.cpp

```diff
@@ -1006,6 +1032,217 @@ namespace RINGMesh
+        const index_t nb_found_surfaces )
+    {
+        std::vector< bool > are_surfaces_hole( nb_found_surfaces, true );
+        for( auto line_id : range( geomodel_.nb_lines() ) )
+        {
+            if( line_2_surface[2 * line_id] != line_2_surface[2 * line_id + 1] )
+            {
+                are_surfaces_hole[line_2_surface[2 * line_id]] = false;
+                are_surfaces_hole[line_2_surface[2 * line_id + 1]] = false;
+            }
+        }
+        index_t nb_pb_surfaces{ static_cast< index_t >( std::count(
+            are_surfaces_hole.begin(), are_surfaces_hole.end(), true ) ) };
+        if( nb_pb_surfaces > 0 )
+        {
+            throw RINGMeshException( "Surface2D", "During surface from corners "
+                                                  " and lines build, ",
+                nb_pb_surfaces,
+                " group(s) of lines are "
+                "floating inside a surface. This is not yet handled by the "
+                "algorithm. Aborting..." );
+        }
+    }
+
+    void GeoModelBuilder< 2 >::build_surface_polygons(
+        const std::vector< std::vector< OrientedLine > >&
+            surface_boundary_lines )
+    {
+        for( const auto& surface_boundaries : surface_boundary_lines )
+        {
+            auto cur_surface_mesh = SurfaceMesh2D::create_mesh();
+            auto cur_surface_mesh_builder =
+                SurfaceMeshBuilder2D::create_builder( *cur_surface_mesh );
+            std::vector< vec2 > polygon_vertices;
+            for( auto cur_surf_boundary : surface_boundaries )
+            {
+                const auto& cur_line =
+                    this->geomodel_.line( cur_surf_boundary.index );
+                for( auto vertex : range( 1, cur_line.nb_vertices() ) )
+                {
+                    cur_surface_mesh_builder->create_vertex( cur_line.vertex(
+                        cur_surf_boundary.side
+                            ? vertex
+                            : ( cur_line.nb_vertices() - 1 ) - vertex ) );
+                    polygon_vertices.push_back( cur_line.vertex(
+                        cur_surf_boundary.side
+                            ? vertex
+                            : ( cur_line.nb_vertices() - 1 ) - vertex ) );
+                }
+            }
+            std::vector< index_t > polygon_corners(
+                cur_surface_mesh->nb_vertices() );
+            std::iota( polygon_corners.begin(), polygon_corners.end(), 0 );
+            //            cur_surface_mesh_builder->create_polygon(
+            //            polygon_vertices );
+            auto surface_id =
+                topology.create_mesh_entity( surface_type_name_static() );
+            std::vector< index_t > polygon_vertex_ptr( 2, 0 );
+            polygon_vertex_ptr[1] =
+                static_cast< index_t >( polygon_corners.size() );
+            geometry.set_surface_geometry( surface_id.index(), polygon_vertices,
```

Comment:
Is there a reason to first create an independent mesh and then set the geometry to surface instead of creating the mesh of the surface and directly modifying it?